src/server/components/dashboard/annualReview/controllers.ts

```diff
@@ -0,0 +1,99 @@
+import * as DateFns from "date-fns";
+import { startCase } from "lodash";
+import { findListById, updateAnnualReviewDate } from "server/models/list";
+import { getCSRFToken } from "server/components/cookies/helpers";
+import { logger } from "server/services/logger";
+import { DEFAULT_VIEW_PROPS } from "server/components/dashboard/controllers";
+import * as Helpers from "server/components/dashboard/annualReview/helpers";
+import { sendAnnualReviewDateChangeEmail } from "server/services/govuk-notify";
+import { dashboardRoutes } from "server/components/dashboard/routes";
+
+import type { NextFunction, Request, Response } from "express";
+import type { List } from "server/models/types";
+
+export const DATE_FORMAT = "d MMMM yyyy";
+
+export async function editDateGetController(req: Request, res: Response, next: NextFunction): Promise<void> {
+  try {
+    const { listId } = req.params;
+    const list = await findListById(listId);
+    const annualReviewStartDate = Helpers.formatAnnualReviewDate(list as List, "annualReviewStartDate");
+    const maxDate = list?.jsonData.annualReviewStartDate ? Helpers.getMaxDate(list) : "";
+    const formattedMaxDate = maxDate ? DateFns.format(maxDate, DATE_FORMAT) : "";
+    const helpText = maxDate
+      ? `The new date must be before ${formattedMaxDate}`
+      : "The new date must be within 6 months of today’s date";
+
+    res.render("dashboard/lists-edit-annual-review-date", {
+      ...DEFAULT_VIEW_PROPS,
+      error: {
+        text: req.flash("annualReviewError")[0],
+      },
+      annualReviewStartDate,
+      maxDate,
+      formattedMaxDate,
+      list,
+      helpText,
+      csrfToken: getCSRFToken(req),
+    });
+  } catch (error) {
+    logger.error(`editDateGetController Error: ${(error as Error).message}`);
+    next(error);
+  }
+}
+
+export async function editDatePostController(req: Request, res: Response, next: NextFunction): Promise<void> {
+  try {
+    return req.body.action === "confirmNewDate"
```

Comment:
req.body.action works better when there are lots of different actions you can do to one resource for example, deleting, editing or adding a user. although I understand that technically you have one route but multiple actions (submitting confirmNewAnnualReview vs updateAnnualReview). 

We can look another time but we probably want to do something like 

`/:listId/settings/users` `/:listId/settings/users/confirm` vs `/:listId/settings/annual-review-date` `/:listId/settings/annual-review-date/confirm` which is more restful.

there's only one thing you can do to review date (change it), but for users you can do multiple.

Reply:
Do you think it's worth making a list and putting this in a refactoring ticket or something?

Reply:
yep. in the backlog